plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/explorer/OpenShiftExplorerContentProvider.java

```diff
@@ -65,7 +65,9 @@ protected void handleConnectionChanged(IConnection connection, String property,
 				cache.flushFor((Connection) newValue);
 			}
 			refreshViewer(newValue);
-		} 
+		} else if (ConnectionProperties.PROPERTY_RESOURCE.equals(property)) {
+		    refreshViewer(newValue);
```

Comment:
this should not be necessary. It used to work when there was no JSON editor.
Now, given a *lot* or resource changes are emitted regularly, that would cause the viewer to flicker a lot